### PR TITLE
Bugfix/1972 monitoring infodate switches 1 day

### PR DESCRIPTION
**Pull request by DzMakatun**
Solves #1972 
- Monitoring now uses controlMeasure.metadata.informationDate for filtering only. The UI displays info_date as an original  string from controlMeasure.metadata.additionalInfo.enceladus_info_date. This way users see the value exactly as specified in the Spark job without unwanted timezone adjustments.
- The filtering on info_date now includes 23:59:59 of the endDate of the interval

### CODE REVIEW

**Review by Zejnilovic — APPROVED** [review]
- Code reviewed
- Built
- Menas Run

File: menas/src/main/scala/za/co/absa/enceladus/menas/repositories/MonitoringMongoRepository.scala [hunk 1]
```diff
@@ -67,7 +67,7 @@ class MonitoringMongoRepository @Autowired()(mongoDb: MongoDatabase)
         Document(
           s"""{ $$match: {
              |    informationDateCasted: {$$gte: ISODate("${startDate}T00:00:00.0Z"),
-             |      $$lte: ISODate("${endDate}T00:00:00.0Z") }
+             |      $$lte: ISODate("${endDate}T23:59:59.9Z") }
```
Comment (benedeki) [inline menas/src/main/scala/za/co/absa/enceladus/menas/repositories/MonitoringMongoRepository.scala:70]: Why this change?

Comment (DzMakatun) [reply menas/src/main/scala/za/co/absa/enceladus/menas/repositories/MonitoringMongoRepository.scala:70]: To include the infodates where time is not set to 00:00:00 UTC into the search results. 
The time is not supposed to be included at all, however, I came across example where the infodate is always 'YYYY-MM-DD 23:59:59.590' ...

File: menas/ui/service/MonitoringService.js [hunk 2]
```diff
@@ -363,8 +363,7 @@ var MonitoringService = new function() {
 
       if (oData.length > 0) {
         for (let oRun of oData) {
-          oRun["infoDate"] = new Date(oRun["informationDateCasted"]["$date"]); // milliseconds to date
-          oRun["infoDateString"] = Formatters.infoDateToString(oRun["infoDate"]);
+          oRun["infoDateString"] = oRun["controlMeasure"]["metadata"]["additionalInfo"]["enceladus_info_date"]
```
[the remark this answers is not in the document]
Comment (DzMakatun) [reply menas/ui/service/MonitoringService.js:366]: Because there is currently no convention on which timezone to use for the infodate I don't see a reliable way to implement this. 

From the other hand, the infodate is mostly treated as a string. So the idea is to "show the user exactly what they entered" and hide the internal transformations. Currently, the cast to ISOdate is only needed to: 1) search the runs 2) sort the runs

[the remark this answers is not in the document]
Comment (DzMakatun) [reply menas/ui/service/MonitoringService.js:366]: More exact: Currently we cast infodate string to ISOdate in mongo query for the search, and we assume everything is in UTC (In case the user uses some other timezones she can adjust her search interval accordingly).  Then we also use it to order the results, which works regardless of the timezone (as long as the same one is used across all the dataset runs). That part remained unchanged. Now the change is in what is displayed in the UI:
- Previously, it was casted back to string from the ISOdate. And therefore, prone to the timezone   shift.
- Now we just show the user exactly the string which she entered herself.

[the remark this answers is not in the document]
Comment (benedeki) [reply menas/ui/service/MonitoringService.js:366]: I am still not convinced correct-correct solution. But as the API will be changing anyway in E3, it can be adjuested then. And until then this is _good enough_ 😉
